src/bin/pg_autoctl/monitor.c

```diff
@@ -416,6 +416,10 @@ monitor_print_nodes_as_json(Monitor *monitor, char *formation, int groupId)
 		log_error("Failed to get the nodes from the monitor while running "
 				  "\"%s\" with formation %s and group %d",
 				  sql, formation, groupId);
+		if (context.strVal)
+		{
+			free(context.strVal);
```

Comment:
I wonder if we should change the definition of SingleValueResultContext (and other similar structs) so that the structure contains a pre-allocated string buffer rather than just a pointer to dynamic memory. Though I must admit fixing what we have now is great, so let's merge that one first and think more later.

Reply:
:+1: